packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Span } from '../../../../types/trace';
+
+// This function turncates/drops the overflowing child spans
+const sanitizeOverFlowingChildren = (spans: Span[]): Span[] => {
+  let sanitizedSpanData: Span[] = [];
+
+  spans.forEach(span => {
+    if (!span.references.length) {
+      sanitizedSpanData.push(span);
+    } else {
+      // parentSpan will be undefined when its parentSpan is dropped previously
+      const parentSpan = sanitizedSpanData.find(s => s.spanID === span.references[0].spanID)!;
+      if (parentSpan) {
+        const childEndTime = span.startTime + span.duration;
+        const parentEndTime = parentSpan.startTime + parentSpan.duration;
+        switch (true) {
+          case span.startTime >= parentSpan.startTime && childEndTime <= parentEndTime:
+            // case 1: everything looks good
+            // |----parent----|
+            //   |----child--|
+            sanitizedSpanData.push(span);
+            break;
+
+          case span.startTime < parentSpan.startTime &&
+            childEndTime <= parentEndTime &&
+            childEndTime > parentSpan.startTime:
+            // case 2: child start before parent, truncate is needed
+            //      |----parent----|
+            //   |----child--|
+            sanitizedSpanData.push({
+              ...span,
+              startTime: parentSpan.startTime,
+              duration: childEndTime - parentSpan.startTime,
+            });
+            break;
+
+          case span.startTime >= parentSpan.startTime &&
+            childEndTime > parentEndTime &&
+            span.startTime < parentEndTime:
+            // case 3: child end after parent, truncate is needed
+            //      |----parent----|
+            //              |----child--|
+            sanitizedSpanData.push({
+              ...span,
+              duration: parentEndTime - span.startTime,
+            });
+            break;
+
+          default:
```

Comment:
But there would be no default case right?? I think comments are mentioning details about what default case handles even then should I write the case instead of default?If yes
Please Clarify this doubt - In eslint rules it is expecting to have a default case in switch block. How to resolve this.

Reply:
I think, you can raise an Exception in the default case, as we don't expect to reach the default case anyway. Better to fail in an explicit way, otherwise debugging errors becomes difficult.

Reply:
Updated it, Let me Know If there are any changes